Erroneous error message printed with `--dryrun`

When testing latest `ZMap` compiled from source with:
`sudo ./src/zmap  -p 80 --dryrun -n 1`

I get this error:
`Jan 21 14:45:19.728 [ERROR] send_batch cleanup: could not clear out remaining batch packets: Undefined error: 0`

This is due to the function `send_batch` not being able to differentiate between:
1.  a batch with packets and none could be sent (error condition)
2. a batch with no packets (not error condition)

Should have caught this in my `send_mmsg`/`batch` testing. Should be an easy fix

--- a/src/send-bsd.h
+++ b/src/send-bsd.h
@@ -41,6 +41,10 @@ int send_packet(sock_t sock, void *buf, int len, UNUSED uint32_t idx)
 // Returns - number of packets sent
 // Returns -1 and sets errno if no packets could be sent successfully
 int send_batch(sock_t sock, batch_t* batch, int retries) {
+	if (batch->len == 0) {
+		// nothing to send
+		return EXIT_SUCCESS;
+	}
 	int packets_sent = 0;
 	int rc = 0;
 	for (int packet_num = 0; packet_num < batch->len; packet_num++) {
@@ -71,8 +75,6 @@ int send_batch(sock_t sock, batch_t* batch, int retries) {
 	if (packets_sent == 0) {
 		// simulating the return behaviour of the Linux send_mmsg sys call on error. Returns -1 and leaves
 		// errno as set by send_packet
-		log_error("send", "send_batch failed and no packets were able to be sent: "
-				  "%s", strerror(errno));
 		return -1;
 	}
 	return packets_sent;

--- a/src/send-linux.c
+++ b/src/send-linux.c
@@ -66,6 +66,10 @@ int send_run_init(sock_t s)
 }
 
 int send_batch(sock_t sock, batch_t* batch, int retries) {
+	if (batch->len == 0) {
+		// nothing to send
+		return EXIT_SUCCESS;
+	}
 	struct mmsghdr msgvec [batch->capacity]; // Array of multiple msg header structures
 	struct msghdr msgs[batch->capacity];
 	struct iovec iovs[batch->capacity];

--- a/src/send.c
+++ b/src/send.c
@@ -429,6 +429,7 @@ int send_run(sock_t st, shard_t *s)
 					int rc = send_batch(st, batch, attempts);
 					// whether batch succeeds or fails, this was the only attempt. Any re-tries are handled within batch
 					if (rc < 0) {
+						log_error("send_batch", "could not send any batch packets: %s", strerror(errno));
 						// rc is the last error code if all packets couldn't be sent
 						s->state.packets_failed += batch->len;
 					} else {

--- a/src/send.c
+++ b/src/send.c
@@ -471,7 +471,7 @@ int send_run(sock_t st, shard_t *s)
 	}
 cleanup:
 	if (!zconf.dryrun && send_batch(st, batch, attempts) < 0) {
-		log_error("send_batch cleanup", "could not clear out remaining batch packets: %s", strerror(errno));
+		log_error("send_batch cleanup", "could not send remaining batch packets: %s", strerror(errno));
 	}
 	free_packet_batch(batch);
 	s->cb(s->thread_id, s->arg);

--- a/src/send.c
+++ b/src/send.c
@@ -469,8 +469,8 @@ int send_run(sock_t st, shard_t *s)
 		}
 	}
 cleanup:
-	if (send_batch(st, batch, attempts) < 0) {
-		perror("error in cleanup, send_batch");
+	if (!zconf.dryrun && send_batch(st, batch, attempts) < 0) {
+		log_error("send_batch cleanup", "could not clear out remaining batch packets: %s", strerror(errno));
 	}
 	free_packet_batch(batch);
 	s->cb(s->thread_id, s->arg);

